examples/bri-3/src/bri/identity/bpiSubjects/models/bpiSubject.ts

```diff
@@ -42,15 +43,18 @@ export class BpiSubject {
     this.description = newDescription;
   }
 
-  public updatePublicKey(newPk: string): void {
-    this.publicKey = newPk;
+  public updatePublicKey(newPk: PublicKey): void {
+    this.publicKey.map((key) => (key.type == newPk.type ? (key = newPk) : key));
```

Comment:
.map would not update the this.publicKey array. it produces a new array.
this.publicKey = this.publicKey.map(...)

Reply:
Fixed.